src/App.js

```diff
@@ -169,6 +170,16 @@ async function makeChatName() {
     return chatName
 }
 
+const partyFromToken = (token) => {
+  try {
+    const decoded = jwt_decode(token);
+    return decoded["https://daml.com/ledger-api"].actAs.shift()
+  } catch (e) {
+    console.log(e.message || "failed to extract party from jwt token")
+    return "";
```

Comment:
maybe best to return `undefined` here. Im not positive but I think the `||` below would settle on `""` instead of defaulting to another value

Reply:
I agree that it's better to return `undefined` than `""`, but because it clarifies that the function returns falsy on exception, though in practice they would both be equivalent in this regard (as would `NaN` or `-0` https://developer.mozilla.org/en-US/docs/Glossary/Falsy).

An expression built of multiple sub expressions combined with the OR operator will evaluate left to right while short circuiting, so ```undefined || '' || 22 || 'zebras'``` still evaluates to ```22``` for instance.